pkg/cmd/cli/secrets/link_secret_to_obj.go

```diff
@@ -0,0 +1,190 @@
+package secrets
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"strings"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	kcmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+
+	"github.com/spf13/cobra"
+)
+
+const (
+	// LinkSecretRecommendedName `oc secrets link`
+	LinkSecretRecommendedName = "link"
+
+	// AddSecretRecommendedName Maintained for backwards-compatibility, but deprecated
+	AddSecretRecommendedName = "add"
+
+	// TODO: move to examples
+	linkSecretLong = `
+Link secrets to a service account
+
+After you have created a secret, you probably want to make use of that secret inside of a pod, for a build, or as an image pull secret.  In order to do that, you must link your secret to a service account.`
```

Comment:
You want to hit three points here:

1. You can auto mount secrets for applications (via the mount) (uncommon)
2. You can add secrets for image pulling (common)
3. Some administrators may require secrets be linked to service accounts (v. uncommon)

So the description needs to reflect that (and the examples should also indicate a realistic example).